contract_show_invoice/views/contract_view.xml

```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0"?>
+<openerp>
+    <data>
+
+        <record id="act_analytic_invoices" model="ir.actions.act_window">
+            <field name="context">{'search_default_analytic_account_ids':
+                [active_id], 'default_analytic_account_ids': active_id}</field>
+            <field name="name">Invoices</field>
+            <field name="res_model">account.invoice</field>
+            <field name="domain">[('analytic_account_ids','=',active_id)]</field>
+            <field name="view_id" ref="account.invoice_tree" />
+            <field name="search_view_id" ref="account.view_account_invoice_filter" />
+        </record>
+
+        <record id="account_analytic_account_button_invoice" model="ir.ui.view">
+            <field name="name">account.analytic.account.button.invoice
+            </field>
+            <field name="model">account.analytic.account</field>
+            <field name="inherit_id" ref="analytic.view_account_analytic_account_form" />
+            <field name="arch" type="xml">
+                <xpath expr='//div[@name="buttons"]' position='inside'>
+                    <button class="oe_stat_button" type="action" icon="fa-edit"
+                        name="%(contract_show_invoice.act_analytic_invoices)d"
+                        string="Invoices" help="Invoices related with this contract">
+                    </button>
```

Comment:
It would be nice for the smart button to include a computed field with the total amount invoiced.
If it's not implemented now, it's a idea to put in the roadmap section.